packages/core/src/components/popover/popover.tsx

```diff
@@ -549,18 +558,17 @@ export class Popover extends AbstractPureComponent2<IPopoverProps, IPopoverState
                 this.props.onInteraction?.(isOpen, e);
             }
             if (!isOpen) {
-                this.props.onClose?.(e);
+                // non-null assertion because the only time `e` is undefined is when in controlled mode
+                // or the rare special case in uncontrolled mode when the `disabled` flag is toggled true
+                this.props.onClose?.(e!);
             }
         }
     }
 
     private isArrowEnabled() {
-        const {
-            minimal,
-            modifiers: { arrow },
-        } = this.props;
+        const { minimal, modifiers } = this.props;
         // omitting `arrow` from `modifiers` uses Popper default, which does show an arrow.
-        return !minimal && (arrow == null || arrow.enabled);
+        return !minimal && modifiers?.arrow?.enabled;
```

Comment:
this looks wrong -- the statements are not functionally equivalent. this can evaluate to `undefined`

Reply:
✅